expression/builtin_miscellaneous_test.go

```diff
@@ -295,7 +299,10 @@ func TestInet6AtoN(t *testing.T) {
 		f, err := fc.getFunction(ctx, datumsToConstants([]types.Datum{ip}))
 		require.NoError(t, err)
 		result, err := evalBuiltinFunc(f, chunk.Row{})
-		require.NoError(t, err)
+		if err != nil {
```

Comment:
You are still using the result `result, err` of `evalBuiltinFunc`to determine if the test check `require.True(t, terror.ErrorEqual(err, errWrongValueForType))` should be executed.

Reply:
Yes, but this is not the same as before, and as you suggested: in err case we must be as we preset expected, not err case also must as we preset expected. So the logic should be same as if (espected ....). I write it this way is because same the code is more concise.

Reply:
But `err` is the result of execution, not the expected error.

Reply:
No. Results can never be the expectation.

Take an  example, I could say "[roly-poly toy](https://en.wikipedia.org/wiki/Roly-poly_toy) will never fail", and that is true. In this case, it is that my expectation happens to be true. Whatever my expectation is, `roly-poly toy` is designed not to fail.

But this is not true on program, where it may be written to a wrong form easily. So you don't know whether the result is correct or not. Thus it will not always match your expectation theoretically, even if *they now match your expectation*.

And we write tests for correctness now, to prevent regression in future. Your argument ignores that *tests will serve for regression*.

Reply:
You are right!!! Sorry and THKS, I'll refix and resubmit

Reply:
Hi, I made new fix, please check. THKS much!

Reply:
Commented.